### PR TITLE
Adding unit test for "Tier0Config"

### PR DESCRIPTION
**Pull request by ItyTell**
Adding unit test for "Tier0Config".

### CODE REVIEW

**Comment by germanfgv** [comment]
@ItyTell looks good, could you please clarify how you are running the tests? I want to reproduce your steps.

**Comment by ItyTell** [comment]
@germanfgv just use "python3 Tier0Config_t.py". (python should have WMCore, T0 and unittest packages installed). I tried it in docker container but should work in any system too.

File: test/python/T0_t/Config/Tier0Config_t.py [hunk 1]
```diff
@@ -0,0 +1,557 @@
...
+        streamConfig = tier0config.retrieveStreamConfig(self.config, nonExistentStreamName)
+
+        # Check if the default values were copied correctly
+        self.assertEqual(streamConfig._internal_name, nonExistentStreamName)
+        self.assertEqual(getattr(self.config.Streams, nonExistentStreamName), streamConfig)
+
+    def test_deleteStreamConfig(self):
+        # Test with an existing stream
+        streamName = "Express"
+        tier0config.deleteStreamConfig(self.config, streamName)
```
Comment (germanfgv) [inline test/python/T0_t/Config/Tier0Config_t.py:32]: AS there was no `Express` stream in the first place, this test would succeed even if the function does not work. We could thest this after testing addRepackConfig, so there is a stream to delete.

File: test/python/T0_t/Config/Tier0Config_t.py [hunk 2]
```diff
@@ -0,0 +1,557 @@
...
+
+        # Check if the stream configuration has been deleted
+        self.assertIsNone(getattr(self.config.Streams, streamName, None))
+
+    def test_retrieveSiteConfig(self):
+        # Test with an existing site
+        siteName = "T1_US_FNAL"
+        siteConfig = tier0config.retrieveSiteConfig(self.config, siteName)
```
Comment (germanfgv) [inline test/python/T0_t/Config/Tier0Config_t.py:40]: The has not been added, therefore it does not really exist. This is similar to the comments about `test_retrieveStreamConfig`

File: test/python/T0_t/Config/Tier0Config_t.py [hunk 3]
```diff
@@ -0,0 +1,557 @@
...
+        # Test with an existing dataset
+        datasetName = "PrimaryDataset"
+        datasetConfig = tier0config.retrieveDatasetConfig(self.config, datasetName)
+        
+        # Test with a non-existing dataset, it should fall back to "Default"
+        nonExistentDatasetName = "NonExistentDataset"
+        datasetConfig = tier0config.retrieveDatasetConfig(self.config, nonExistentDatasetName)
+
+        # Check if the default values were copied correctly
+        self.assertEqual(datasetConfig._internal_name, nonExistentDatasetName)
```
Comment (germanfgv) [inline test/python/T0_t/Config/Tier0Config_t.py:68]: Same as `test_retrieveStreamConfig`

**Review by germanfgv — COMMENTED** [review]
Other than the issues I have with the first group of unit tests, I thing this should work prety well. 

Another detail: Please change the path from `test/python/T0_t/Config/Tier0Config_t.py` to `test/python/T0_t/RunConfig_t/Tier0Config_t.py`

**Review by germanfgv — APPROVED** [review]
Much better. Thank you @ItyTell

**Pull request merged**